src/main/java/com/ericsson/gerrit/plugins/eiffel/listeners/AbstractEventListener.java

```diff
@@ -146,6 +164,13 @@ protected boolean isEiffelEventSendingEnabled(final Event gerritEvent,
     protected abstract void prepareAndSendEiffelEvent(Event gerritEvent,
             EiffelPluginConfiguration pluginConfig);
 
+    private void initializeThreadPoolExecutor() {
+        if (initialized.compareAndSet(false, true)) {
+            executor = new ThreadPoolExecutor(CORE_POOL_SIZE, MAXIMUM_POOL_SIZE, KEEP_ALIVE,
```

Comment:
Doesn't this executor have to instantiated by injection? I can't see how this would continue to operate after we have exited the thread or am I missing something?

Reply:
The executor is static so its shared among any instance of the abstract listener which is extended by the other listeners. The executor itself is not a thread but is like a "worker" manager. When you submit something to the queue one of the available thread slots will be taken. In this case I've set it to a max of 3 simultaneous threads active.

Reply:
So the thread that is started in this thread willl not be killed when this thread dies?

Reply:
As I undertsnd it it is the Executor class that starts the thread and Gerrit is responsible of the executor class.